src/main/java/org/example/repository/CustomerRepository.java

```diff
@@ -2,24 +2,94 @@
 
 import org.example.entity.Customer;
 
-public class CustomerRepository implements CrudRepository<Customer>{
+
+import java.sql.*;
+
+
+public abstract class CustomerRepository implements CrudRepository<Customer> {
+
+    private final String url = "jdbc:postgresql://localhost:5432/docker";
+    private final String user = "docker";
+    private final String password = "docker";
+
     @Override
     public Customer findById(int id) {
+        String FIND_BY_ID_QUERY = "SELECT * FROM customer WHERE id = ?";
+        try (Connection connection = DriverManager.getConnection(url, user, password);
```

Comment:
Don't use the DriverManager here, you have a separate class for this (DbConnection)